### PR TITLE
1) Created calculator, added test for all operation, create custom Ex…

### PR DESCRIPTION
**Pull request by AntonPanchishko**
…ception

### CODE REVIEW

File: src/main/java/core/basesyntax/Calculator.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            case '*':
+                return first == 0 || second == 0 ? 0 : (double) first * second;
+            case '/':
+                if (second == 0) {
+                    throw new ArithmeticException("Sorry, I can't divide on zero");
+                }
+                return (double) first / second;
+            case '^':
+                if (first == 0 && second > 0) {
+                    throw new ArithmeticException("Sorry, I can't raising zero in negative pow");
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/Calculator.java:20]: I think u can raise zero in any power

File: src/main/java/core/basesyntax/Calculator.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+public class Calculator {
+    public double calculate(double first, char operation, double second) {
+
+        switch (operation) {
+            case '+':
+                return (double) first + second;
+            case '-':
+                return (double) first - second;
+            case '*':
+                return first == 0 || second == 0 ? 0 : (double) first * second;
```
Comment (Max-Pochepets) [inline src/main/java/core/basesyntax/Calculator.java:12]: do we need a check?

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class CalculatorTest {
+    private Calculator calculator = new Calculator();
```
Comment (Max-Pochepets) [inline src/test/java/core/basesyntax/CalculatorTest.java:9]: let's not create a field, do it with `Test methods`

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]
👍

File: src/main/java/core/basesyntax/Calculator.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+public class Calculator {
+    public double calculate(double first, char operation, double second) {
+
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Calculator.java:5]: redundant empty line

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class CalculatorTest {
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:9]: let's add access modifiers to the class and to all its methods

File: src/main/java/core/basesyntax/Calculator.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        switch (operation) {
+            case '+':
+                return (double) first + second;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Calculator.java:8]: do we need casting in all cases?

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            calculator.calculate(-1000, 'ж', 1000);
+        });
+    }
+
+    @Test
+    void addiction_Ok() {
+        assertEquals(2000, calculator.calculate(1000, '+', 1000));
+        assertEquals(-2000, calculator.calculate(-1000, '+', -1000));
+        assertEquals(0, calculator.calculate(1000, '+', -1000));
+        assertEquals(0, calculator.calculate(-1000, '+', 1000));
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:44]: let's also check passing 0 as one of the operands

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        assertEquals(2.147484647E9, calculator.calculate(Integer.MAX_VALUE, '+', 1000));
+        assertEquals(-2.147482648E9, calculator.calculate(Integer.MIN_VALUE, '+', 1000));
+    }
+
+    @Test
+    void subtraction_Ok() {
+        assertEquals(0, calculator.calculate(1000, '-', 1000));
+        assertEquals(0, calculator.calculate(-1000, '-', -1000));
+        assertEquals(2000, calculator.calculate(1000, '-', -1000));
+        assertEquals(-2000, calculator.calculate(-1000, '-', 1000));
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:54]: same

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Test
+    void multiplication_Ok() {
+        assertEquals(4, calculator.calculate(2, '*', 2));
+        assertEquals(4, calculator.calculate(-2, '*', -2));
+        assertEquals(-4, calculator.calculate(2, '*', -2));
+        assertEquals(-4, calculator.calculate(-2, '*', 2));
+        assertEquals(0, calculator.calculate(0, '*', 2));
+        assertEquals(0, calculator.calculate(-2, '*', 0));
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:66]: what about edge values?

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Test
+    void division() {
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:70]:
```suggestion
    void division_Ok() {
```

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    void raisingToPower_Ok() {
+        assertEquals(4, calculator.calculate(2, '^', 2));
+        assertEquals(0.25, calculator.calculate(-2, '^', -2));
+        assertEquals(0.25, calculator.calculate(2, '^', -2));
+        assertEquals(4, calculator.calculate(-2, '^', 2));
+
+        assertEquals(4.572473708276177E-4, calculator.calculate(3, '^', -7));
+        assertEquals(1.0861771343660416E31, calculator.calculate(43, '^', 19));
+        assertEquals(-4913.0, calculator.calculate(-17, '^', 3));
+        assertEquals(-3.535942135457617E-18, calculator.calculate(-22, '^', -13));
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:97]: let's add checks of raising to zero power
and raising 0 to some positive power

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Test
+    void raisingToPower_NotOK() {
+        assertThrows(ArithmeticException.class, () -> {
+            calculator.calculate(2000000000, '^', 2000000000);
+            assertThrows(ArithmeticException.class, () ->
+                    calculator.calculate(0, '^', -4));
+        });
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:106]: one check is inside of the other

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+        assertEquals(1, calculator.calculate(-4, '/', -4));
+        assertEquals(-1, calculator.calculate(4, '/', -4));
+        assertEquals(-1, calculator.calculate(-4, '/', 4));
+        assertEquals(2.5, calculator.calculate(5, '/', 2));
+        assertEquals(5.666666666666667, calculator.calculate(-17, '/', -3));
+        assertEquals(-49.75, calculator.calculate(199, '/', -4));
+        assertEquals(-5.2631578947368425, calculator.calculate(-1, '/', 0.19));
+
+        assertEquals(0, calculator.calculate(0, '^', 3));
+        assertEquals(1, calculator.calculate(3, '^', 0));
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:87]: I think this belongs to a different test?

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+        assertEquals(1.0861771343660416E31, calculator.calculate(43, '^', 19));
+        assertEquals(-4913.0, calculator.calculate(-17, '^', 3));
+        assertEquals(-3.535942135457617E-18, calculator.calculate(-22, '^', -13));
+    }
+
+    @Test
+    public void raisingToPower_NotOK() {
+        assertThrows(ArithmeticException.class, () -> {
+            calculator.calculate(2000000000, '^', 2000000000);
+        });
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:112]:
```suggestion
        assertThrows(ArithmeticException.class, () -> 
            calculator.calculate(2000000000, '^', 2000000000);
        );
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
added one comment

**Review by KseniiaMakarova — APPROVED** [review]
